src/modules/SerialModule.cpp

```diff
@@ -198,8 +199,157 @@ int32_t SerialModule::runOnce()
                     }
                 }
             }
+
 #if !defined(TTGO_T_ECHO) && !defined(CANARYONE)
-            else {
+            else if ((moduleConfig.serial.mode == meshtastic_ModuleConfig_SerialConfig_Serial_Mode_WS85)) {
+                static unsigned int lastAveraged = 0;
```

Comment:
Given the extensive parsing logic and math involved, I wonder if we should make a void function to extract this block to, so that it doesn't bloat runOnce()

Reply:
Sure I can do that.  All that logic is pretty confusing.